yaksh/static/yaksh/js/requesthandler.js

```diff
@@ -0,0 +1,139 @@
+request_status = "initial"
+count = 0;
+checker = null
+function submitRequest(){
+    document.forms["code"].submit();
+}
+
+function check_state(state, uid) {
+    if ((state == "running" || state == "not started") && count < 7) {
+        count++;
+        setTimeout(function() {get_result(uid);}, 2000);
+    } else if (state == "unknown") {
+        request_status = "initial";
+        count = 0;
+        notify("Request timeout. Try again later");
+        clearInterval(checker);
+        unlock_screen();
+    } else {
+        request_status = "initial";
+        count = 0;
+        notify("Please try after few minutes");
+        clearInterval(checker);
+        unlock_screen();
+    }
+}
+
+function notify(text) {
+    var $notice = document.getElementById("notification");
+    $notice.classList.add("alert");
+    $notice.classList.add("alert-success");
+    $notice.innerHTML = text;
+}
+
+function lock_screen() {
+    document.getElementById("ontop").style.display = "block";
+}
+
+function unlock_screen() {
+    document.getElementById("ontop").style.display = "none";
+}
+
+function check_lock_screen() {
+    var $ontop_div = document.getElementById("ontop");
+    if ($ontop_div.style.display == "block") {
+        $ontop_div.style.display = "none";
+    }
+}
+
+function get_result(uid){
+    var url = "/exam/get_results/"+uid+"/";
+    ajax_call(url, "GET", "html", null, uid)
+}
+
+function ajax_call(url, method_type, data_type, data, uid) {
```

Comment:
I don't like the name `ajax_call`, why is it not descriptive of what it is doing?  For example `ajax_get_code_status` or something?

Reply:
Actually,  post and get request are made via same function so kept name generic.
I have changed to ajax_check_code as ultimately the user sees the error or next question if code is correct.